challenges.js

```diff
@@ -115,3 +115,45 @@ const challengesContainer = document.querySelector(".challenges-container");
 
 let view = new ChallengeListView();
 view.render(challengesContainer);
+
+//function keyword filter
+
+function keyFilter() {
+  // Declare variables
+  var input, filter, challenges, title, infoText, i;
+  input = document.getElementById('textFilter');
+  if (!input) {
+    console.error('Input element not found.!');
+    return;
+  }
+  filter = input.value.toUpperCase();
+  challenges = document.querySelectorAll('.challenges-container__challenge');
+
+  // loop through challenges
+  for (i = 0; i < challenges.length; i++) {
+    /* console.log('Challenge:', challenges[i]); */
+    title = challenges[i].querySelector(
+      '.challenges-container__challenge__title'
+    );
+    /* console.log(title); */
+    infoText = challenges[i].querySelector(
+      '.challenges-container__challenge__text'
+    );
+    /* console.log(infoText); */
+    if (title && infoText) {
+      const titleText = title.textContent || title.innerHTML;
+      const textContent = infoText.textContent || infoText.innerText;
+
+      if (
+        titleText.toUpperCase().indexOf(filter) > -1 ||
+        textContent.toUpperCase().indexOf(filter) > -1
```

Comment:
Comparing strings using `indexOf()` is also quite oldschool. A more common way today is to simply use the `String.prototype.includes()` function, which is IMO easier to read, and also less error prone (because it returns a boolean you don't have to remember to compare with `-1`).